events/consumer.go

```diff
@@ -112,17 +117,165 @@ func (sr *instrumentedReader) Read(p []byte) (int, error) {
 	return n, err
 }
 
+// Options that control how we should subscribe to a remote event stream
+type HandleRepoStreamRobustOptions struct {
+	// Controls how work should be processed when reading from the remote
+	Scheduler Scheduler
+
+	// The full URL to which we will connect (i.e. `wss://example.com:8080/xrpc/com.atproto.sync.subscribeRepos`)
+	URL string
+
+	// The cursor to use when dialing the remote repo (optional)
+	Cursor *int
+
+	// The dialer user to establish the websocket connection
+	Dialer *websocket.Dialer
+
+	// The list of HTTP headers to pass when establishing the websocket connection. If users do
+	// not supply a `User-Agent`, a default will be provided.
+	Header http.Header
+
+	// The maximum number of consecutive reconnects to try before shutting down
+	MaxReconnectAttempts int
+
+	// Uses the default logger if none is provided
+	Log *slog.Logger
+}
+
+// Constructs an options object with a set of sane default arguments that can be overridden by the user
+func DefaultHandleRepoStreamRobustOptions(sched Scheduler, u string) *HandleRepoStreamRobustOptions {
+	return &HandleRepoStreamRobustOptions{
+		Scheduler:            sched,
+		URL:                  u,
+		Dialer:               websocket.DefaultDialer,
+		Header:               http.Header{},
+		MaxReconnectAttempts: 10,
+		Log:                  slog.Default().With("system", "events"),
+	}
+}
+
+// The same as `HandleRepoStream`, but with auto-reconnects in the case of upstream disconnects
+func HandleRepoStreamRobust(ctx context.Context, opts *HandleRepoStreamRobustOptions) error {
```

Comment:
I think it might make sense since we've only got 2 required params to do something like this?
```
func HandleRepoStreamRobust(ctx context.Context, sched Scheduler, url string, opts *HandleRepoStreamRobustOptions) error {
```